src/test/com/aca/homework/week9/invoice/service/InvoiceServiceTest.java

```diff
@@ -0,0 +1,65 @@
+package com.aca.homework.week9.invoice.service;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.FileNotFoundException;
+
+class InvoiceServiceTest {
+    private InvoiceService testSubject;
+
+    @BeforeEach
+    public void setUp() throws FileNotFoundException {
+        testSubject = new InvoiceService();
+        testSubject.load("invoices.txt");
+    }
+
+    @Test
+    public void testTotalAmountByTypeSAD() {
+        Assertions.assertEquals(2602704070L, testSubject.totalAmountByType(InvoiceType.SAD));
+    }
+
+    @Test
+    public void testGetBySADType() {
+        Invoice[] filteredBySADType = testSubject.getByType(InvoiceType.SAD);
+        boolean allTypesAreSAD = true;
+        for (Invoice invoice : filteredBySADType) {
+            if (invoice.getType() != InvoiceType.SAD) {
+                allTypesAreSAD = false;
+            }
+        }
+        Assertions.assertTrue(allTypesAreSAD);
+    }
+
+    @Test
+    public void testGetByMANType() {
+        Invoice[] filteredByMANType = testSubject.getByType(InvoiceType.MAN);
+        boolean allTypesAreMAN = true;
+        for (Invoice invoice : filteredByMANType) {
+            if (invoice.getType() != InvoiceType.MAN) {
+                allTypesAreMAN = false;
+                break;
+            }
+        }
+        Assertions.assertTrue(allTypesAreMAN);
+    }
+
+    @Test
+    public void getBySADTypeAndCode() {
+        Invoice[] filteredByTypeAndCode = testSubject.getByTypeAndCode(InvoiceType.SAD, "MY-CODE");
+        boolean allTypesAreSAD = true;
+        boolean allCodesAreMyCode = true;
+        for (Invoice invoice : filteredByTypeAndCode) {
```

Comment:
please do not write a duplicate code in your test. just invoke the getByType method and assert that the result is exactly what you expect

Reply:
But what is the result that I expect? I thought the most appropriate assertion would be to check if all the invoices in the array returned by the method getByTypeAndCode have the type SAD and code "MY-CODE".

Reply:
a better approach is to not read real invoices from the .txt file, instead, load for instance 5 manually created invoices and assert that the result contains for instance 2 invoices. it will be more readable and more obvious